### PR TITLE
fix(cpu-template-helper): Fix error message display

### PR DESCRIPTION
**Pull request by zulinx86**
## Changes

- Don't show diff twice.
- Handle a new line correctly in error message.

## Reason

Before
```
$ build/cargo_target/x86_64-unknown-linux-musl/debug/cpu-template-helper fingerprint compare --prev tests/data/cpu_template_helpe
r/fingerprint_INTEL_SKYLAKE_5.10host.json --curr tests/data/cpu_template_helper/fingerprint_INTEL_SKYLAKE_6.1host.json 
Difference detected between source and target: \n{
  "name": "kernel_version",
  "prev": "5.10.215-203.850.amzn2.x86_64",
  "curr": "6.1.84-99.169.amzn2023.x86_64"
}
Error: FingerprintCompare(DiffDetected("{\n  \"name\": \"kernel_version\",\n  \"prev\": \"5.10.215-203.850.amzn2.x86_64\",\n  \"curr\": \"6.1.84-99.169.amzn2023.x86_64\"\n}"))
```

After
```
$ build/cargo_target/x86_64-unknown-linux-musl/debug/cpu-template-helper fingerprint compare --prev tests/data/cpu_template_helper/fingerprint_INTEL_SKYLAKE_5.10host.json --curr tests/data/cpu_template_helper/fingerprint_INTEL_SKYLAKE_6.1host.json 
Difference detected between source and target:
{
  "name": "kernel_version",
  "prev": "5.10.215-203.850.amzn2.x86_64",
  "curr": "6.1.84-99.169.amzn2023.x86_64"
}
```

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this
  PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- ~~[ ] User-facing changes are mentioned in `CHANGELOG.md`.~~
- [x] All added/changed functionality is tested.
- ~~[ ] New `TODO`s link to an issue.~~
- [x] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: src/cpu-template-helper/src/main.rs [hunk 1]
```diff
@@ -190,14 +190,14 @@ fn run(cli: Cli) -> Result<(), HelperError> {
     Ok(())
 }
 
-fn main() -> Result<(), HelperError> {
+fn main() {
```
Comment (roypat) [inline src/cpu-template-helper/src/main.rs:193]:
```suggestion
fn main() -> std::process::ExitCode {
```

File: src/cpu-template-helper/src/main.rs [hunk 2]
```diff
@@ -190,14 +190,14 @@ fn run(cli: Cli) -> Result<(), HelperError> {
...
     let cli = Cli::parse();
     let result = run(cli);
     if let Err(e) = result {
         eprintln!("{}", e);
-        Err(e)
+        std::process::exit(1);
```
Comment (roypat) [inline src/cpu-template-helper/src/main.rs:198]:
```suggestion
        std::process::ExitCode::FAILURE
```

File: src/cpu-template-helper/src/main.rs [hunk 3]
```diff
@@ -190,14 +190,14 @@ fn run(cli: Cli) -> Result<(), HelperError> {
...
     } else {
-        Ok(())
+        std::process::exit(0);
```
Comment (roypat) [inline src/cpu-template-helper/src/main.rs:200]:
```suggestion
        std::process::ExitCode::SUCCESS
```

**Review by roypat — COMMENTED** [review]
I think generally in Rust, one should avoid using `std::process::exit` (https://stackoverflow.com/questions/39228685/when-is-stdprocessexit-o-k-to-use) because it interferes with dropping and unwinding. I don't think it really matters here, because its the very end of `main`, but maybe it's still nice to return `ExitCode` instead, just in case?

**Comment by zulinx86** [comment]
> I think generally in Rust, one should avoid using `std::process::exit` (https://stackoverflow.com/questions/39228685/when-is-stdprocessexit-o-k-to-use) because it interferes with dropping and unwinding. I don't think it really matters here, because its the very end of `main`, but maybe it's still nice to return `ExitCode` instead, just in case?

I really love it. will apply thanks!

**Review by pb8o — APPROVED** [review]

**Pull request merged**